src/PlaywrightSharp.Abstractions/IPage.cs

```diff
@@ -264,9 +276,34 @@ public interface IPage : IDisposable
         /// <param name="e">Event to wait for.</param>
         /// <param name="options">Extra options.</param>
         /// <typeparam name="T">Return type.</typeparam>
+        /// <example>
+        /// <code>
+        /// <![CDATA[
+        /// // wait for console event:
+        /// var console = await page.WaitForEvent<ConsoleEventArgs>(PageEvent.Console);
+        ///
+        /// // wait for popup event:
+        /// var popup = await page.WaitForEvent<PopupEventArgs>(PageEvent.Popup);
+        ///
+        /// // wait for dialog event:
+        /// var dialog = await page.WaitForEvent<DialogEventArgs>(PageEvent.Dialog);
+        ///
+        /// // wait for request event:
+        /// var request = await page.WaitForEvent<RequestEventArgs>(PageEvent.Request);
+        /// ]]>
+        /// </code>
+        /// </example>
         /// <returns>A <see cref="Task"/> that completes when the predicate returns truthy value. Yielding the information of the event.</returns>
         Task<T> WaitForEvent<T>(PageEvent e, WaitForEventOptions options = null);
 
+        /// <summary>
+        /// Waits for event to fire once and return its value.
+        /// </summary>
+        /// <typeparam name="T">Return type.</typeparam>
+        /// <param name="e">Event to wait for.</param>
+        /// <returns>A <see cref="Task"/> that completes when the predicate returns truthy value. Yielding the information of the event.</returns>
+        Task<T> OnceAsync<T>(PageEvent e);

```

Comment:
@kblok since we have the `WaitForEvent` does this make sense?

Reply:
This shouldn't be part of the API.